lib/modules/manager/vendir/extract.ts

```diff
@@ -0,0 +1,103 @@
+import is from '@sindresorhus/is';
+import { logger } from '../../../logger';
+import { parseSingleYaml } from '../../../util/yaml';
+import { HelmDatasource } from '../../datasource/helm';
+import { getDep } from '../dockerfile/extract';
+import { isOCIRegistry, resolveAlias } from '../helmv3/utils';
+import type {
+  ExtractConfig,
+  PackageDependency,
+  PackageFileContent,
+} from '../types';
+import type { HelmChart, Vendir } from './types';
+import { isHelmChart } from './utils';
+
+// TODO: Add support for other vendir types (like git tags, github releases, etc.)
+// Recommend looking at the kustomize manager for more information on support.
+
+export function extractHelmChart(
+  helmChart: HelmChart,
+  aliases?: Record<string, string> | undefined,
+): PackageDependency | null {
+  if (isOCIRegistry(helmChart.repository.url)) {
+    const dep = getDep(
+      `${helmChart.repository.url.replace('oci://', '')}/${helmChart.name}:${helmChart.version}`,
+      false,
+      aliases,
+    );
+    return {
+      ...dep,
+      depName: helmChart.name,
+      packageName: dep.depName,
+      // https://github.com/helm/helm/issues/10312
+      // https://github.com/helm/helm/issues/10678
+      pinDigests: false,
+    };
+  }
+  const repository = resolveAlias(helmChart.repository.url, aliases!);
```

Comment:
normal helm charts don't need registry aliases, so revert. those can be changed by simply override registry url. so aliases are only needed for oci charts.

Reply:
I just mirrored what the kustomize manager had... They support aliases for both oci and regular charts. Which is what you had pointed me to. Do we not want to keep those consistent?

Reply:
oh, didn't know that. I only added aliases for oci charts recently. 🤔

so yes, leave it consistent. please update the registry aliases config option and add vendir there.

Reply:
and don't add the `!` this will cause runtime failures